### PR TITLE
refactor: Remove unwrap and handle errors

### PR DESCRIPTION
**Pull request by jim4067**
References Issue #174

Remove `unwraps` by returning `Result`

### CODE REVIEW

**Comment by aviramha** [comment]
Thanks for your PR! Please notice that the tests failed - you need to change the call to the function to unwrap/pass the error along, also need to run `cargo fmt --all`

**Comment by jim4067** [comment]
okay. let me look into it.

**Comment by jim4067** [comment]
Is there a way to run these tests locally? Also rust-analyzer fails on my machine. I am on Linux, pop os

**Comment by aviramha** [comment]
> Is there a way to run these tests locally? Also, rust-analyzer fails on my machine. I am on Linux, pop os

Yes! We just released out local test/development guide - https://mirrord.dev/docs/developer/testing/
I haven't tested PopOS but requirements are mostly having a K8S cluster and a Unix system.

File: mirrord-layer/src/pod_api.rs [hunk 1]
```diff
@@ -110,6 +105,7 @@ pub async fn create_agent(config: LayerConfig) -> Result<Portforwarder, LayerErr
             &impersonated_pod_namespace,
             &impersonated_container_name,
         )
+        .unwrap()
         .await;
```
Comment (aviramha) [inline mirrord-layer/src/pod_api.rs:109]: ah, no need to unwrap as you can just propogate, sorry about the bad advice.
try
`.await?;` instead

Comment (jim4067) [reply mirrord-layer/src/pod_api.rs:109]: no worries. I am also flying in the dark here.

**Comment by aviramha** [comment]
P.S regardless of tests please add an entry to the CHANGELOG file

**Comment by aviramha** [comment]
@jim4067 I think you should try to get clippy/fmt to work locally as it might be impossible to have the issue solved current way :) If you need any help you can talk with us on the Discord channel.

**Comment by jim4067** [comment]
@aviramha okay. let me check it out.

**Comment by jim4067** [comment]
@aviramha Found where I had messed up. I was using `?` on `Option` when it should be used on `Result`

**Comment by aviramha** [comment]
Hey @jim4067, are you planning on completing this PR? Feel free to get support on Discord if you need any help.

**Review by aviramha — APPROVED** [review]

**Pull request merged**